### PR TITLE
BUG/TST: added curate_streams to zip + other

### PR DESCRIPTION
**Pull request by ordirules**
* fixed curate_streams to work in a nested fashion
    * added a test for curate_streams intended behaviour
    * added curate_streams before emitting on zip

Discussion from #45 
Also, see tests as a point of discussion as to how `curate_streams` should behave. I made a slight line change to allow for `curate_streams(curate_streams(nds))` to work properly.

### CODE REVIEW

**Comment by ordirules** [comment]
(rebased)

File: shed/event_streams.py [hunk 1]
```diff
@@ -250,7 +250,9 @@ def curate_streams(self, nds):
         else:
             names, docs = nds
             name = names
-            docs = (docs,)
+            # make sure it's a dict not tuple
+            if not isinstance(nds[1], tuple):
```
Comment (ordirules) [inline shed/event_streams.py]: @CJ-Wright in particular, i think this should allow nested calls of `curate_streams`

File: shed/event_streams.py [hunk 2]
```diff
@@ -735,7 +737,7 @@ def update(self, x, who=None):
             tup = tuple(buf.popleft() for buf in self.buffers)
             self.condition.notify_all()
             self.prior = tup
-            return self.emit(tup)
+            return self.emit(self.curate_streams(tup))
```
Comment (CJ-Wright) [inline shed/event_streams.py]: We could decorate emit with `self.curate_streams` such that all emitted documents are within our event model.

Comment (ordirules) [inline shed/event_streams.py]: I think we'll need to think more carefully on that one. Simply decorating may not work, since some nodes assume that nds is `name, doc` whereas others expect `name, docs` where `docs` is a tuple (even if length 1).
Maybe it is good to make it another PR. Normalizing all the inputs/outputs of the streams altogether.
It could supersede this one (so we don't merge this)

Comment (CJ-Wright) [inline shed/event_streams.py]: We should demand that all data is `(name, (docs, ...))` at the boundaries of all the nodes.

Comment (ordirules) [inline shed/event_streams.py]: agreed. i guess i could just modify this PR a little later as well, unless you'd rather take care of this one. just let me know!

**Comment by CJ-Wright** [comment]
Superseded by #55
